.flake8

```diff
@@ -12,4 +12,4 @@ ignore=
     ANN102, # cls param annotation
     ANN204, # return type annotation for special methods
     B024,   # ABC class without any abstractmethods
-per-file-ignores=test_*.py:ANN
+per-file-ignores=tests/*:ANN
```

Comment:
I added `__init__.py` file, which contain abstract tests. So should I undo a change in `.flake8` and use types inside tests or what?

Reply:
However, I deleted deprecation system in 49b7382d319f942c0fbddc961fd69b4dc80773ba, so reverted this change in c2ef9a1ccfb5f0ebf3433dc4bc7ba55dd964422f.

Reply:
Revert this revert in 56c85f5732f145cace86d220b6b2d4e1f0410a2a because there is still `__init__.py` file.

Reply:
Sure, there can be an `__init__.py` file, but that doesn't mean we need to ignore annotations in it. Again, I only did this for test files is because we only want to ignore annotation rules in test functions, not in some shared helper functions/classes, we should enforce annotations in those just like everywhere else. However in tests, due to patches and the constant `-> None`, I've made it an exception.

Unless that `__init__.py` contains actual tests, it shouldn't be excepted, and if it does, why?

Reply:
It contains because it then pastes implementations of the tests in actual class.

However, I remember that can ignore part of file. Will try to look at it later.

Reply:
Fixed in d4f302352ad5dbe6b8e40b2e62a0417265399302.